docs/api.md

```diff
@@ -558,6 +559,58 @@ Response:
 }
 ```
 
+### `GET /api/saved-pages`
+
+Returns all a user's saved pages
+
+#### Request headers
+
+- `GOVUK-Account-Session`
+  - the user's session identifier
+
+#### JSON response fields
+
+- `govuk_account_session` *(optional)*
+  - a new session identifier
+- `saved_pages`
+  - an array of pages the user has saved, identified by their page path
+
+#### Response codes
+
+- 401 if the session identifier is invalid
+- 200 otherwise
+
+#### Example request / response
+
+Request (with gds-api-adapters):
+
+```ruby
+GdsApi.saved_page_api.get_saved_pages(
+    govuk_account_session: "session-identifier",
+)
+```
+
+Response when no pages are saved:
+
+```json
+{
+    "govuk_account_session": "YWNjZXNzLXRva2Vu.cmVmcmVzaC10b2tlbg==",
+    "saved_pages": []
+}
+```
+
+Response when a user has saved two pages:
+
+```json
+{
+    "govuk_account_session": "YWNjZXNzLXRva2Vu.cmVmcmVzaC10b2tlbg==",
+    "saved_pages": [
+      { "page_path": "%2Fgovernment%2Forganisations%2Fgovernment-digital-service" },
+      { "page_path": "%2Fgovernment%2Forganisations%2Fcabinet-office" },
```

Comment:
As mentioned in the previous commit, we should be working with actual paths, not escaped ones.  This is introducing potential points of confusion and bugs.

For example, the attribute service prototype expects attribute values to be passed around as encoded JSON strings.  We had an issue where [those strings weren't being unencoded](https://github.com/alphagov/govuk-attribute-service-prototype/pull/112), and so new users who signed up weren't able to return to their saved results.  If we'd just never had the additional encoding step, that problem wouldn't have occurred.